report/src/main/webapp/app/components/keyboardShortcuts.directive.js

```diff
@@ -208,13 +226,54 @@ define(['angularAMD', 'userSettingsService'], function (angularAMD) {
         }
       }
 
-      function toggleNextTest(currentTest) {
-        $(currentTest).nextAll('.aside-report:not(.is-hidden)')
-        .first()
-        .find('.test-name')
-        .click();
+      function toggleNextTestItem(currentTest) {
+        $(currentTest).nextAll(
+          '.aside-report:not(.is-hidden)')
+          .first()
+          .find('.test-url')
+          .first()
+          .click();     
       }
 
+      function togglePrevTestItem(currentTest) {
+        $(currentTest).prevAll(
+          '.aside-report:not(.is-hidden)')
+          .first()
+          .find('.test-url')
+          .last()
+          .click();
+      }
+
+      function clickOnTab() {
+        var currentTab = userSettingsService.getLastTab();
+        var nextTestTabs = $('.nav-tabs').children();
+        for(var i=0;i <nextTestTabs.length; i++) {
+          if($(nextTestTabs[i]).text().replace(/\s/g, '') === currentTab) {
+            $(nextTestTabs[i]).find('a').click();
+            currentTab = null;
+          }
+        }
+      }
+
+      var mutationObserver = new MutationObserver(callback);
```

Comment:
Can you shortly explain why it's needed with config set to `true` for each property?

Reply:
I've used this: https://developer.mozilla.org/en-US/docs/Web/API/MutationObserverInit 
I've tried few configs and this one has been working properly for me.

Reply:
Ok, but why do you need MutationObserver, whatever its config is?

Reply:
It looks for changes in the DOM before it is displayed to the user.

Here, if I didn't use MutationObserver my screen would flick every time I changed tree URL, because  selecting the tab is done by using .click(); 

With MutationObserver I can atually click something before it is displayed (but it already exists in the DOM), both the delay and the flickering is gone.

I've tried using timeout - but it is based on how many tests are performed, the more tests, the longer the timeout should be (and it varies from 10 to 50ms +). Then I've tried with window.onhashchange but it didn't change anything. I just needed something that would detect a change before user sees anything.

Reply:
It's worth to keep in mind that MutationObserver with such an extensive config might be performance heavy. It also sounds to me that it's used to fix a problem that should not occur - I don't think that screen should flick every time you change the URL. Maybe it would be better to spend more time on it and fix it properly.
I would suggest at least adding a short comment explaining why MutationObserver is used so future developers working on it have a clear understanding.